numpy/core/src/multiarray/ctors.c

```diff
@@ -1809,8 +1815,9 @@ PyArray_GetArrayParamsFromObject(PyObject *op,
 
     /* Try to treat op as a list of lists or array-like objects. */
     if (!writeable && PySequence_Check(op)) {
-        int check_it, stop_at_string, stop_at_tuple, is_object;
+        int check_it, stop_at_string, stop_at_tuple;
         int type_num, type;
+        discovered_t is_object = DISCOVERED_OK;
```

Comment:
Can this move down to line 1862, now that we have C99?

Reply:
fixing